drivers: eeprom: stm32: HAL return value handling

etienne-lms opened the pull request:
Correct `eeprom_stm32_write()` to return a valid _errno_ instead of mixing HAL return values and _errno_ return values.

Clarify HAL return value is of type `HAL_StatusTypeDef` and not an `int` in `eeprom_stm32_read()`.

etienne-lms committed:
drivers: eeprom: stm32: don't mix HAL return value and errno

Correct eeprom_stm32_write() to return a valid errno instead
of mixing HAL return values and errno return values.

Clarify HAL return value is of type HAL_StatusTypeDef and not an
int in eeprom_stm32_read().

Remove printing of HAL_FLASHEx_DATAEEPROM_Lock() error code since
not very useful.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

erwango commented on drivers/eeprom/eeprom_stm32.c:
```diff
@@ -70,14 +70,20 @@ static int eeprom_stm32_write(const struct device *dev, off_t offset,
+			ret = HAL_FLASHEx_DATAEEPROM_Lock();
+			if (ret != HAL_OK) {
+				LOG_ERR("failed to lock to EEPROM (err %d)", ret);
+			}
```
`HAL_FLASHEx_DATAEEPROM_Lock` only implemented on L0/L1 and does not perform error checking.
I'm fine to perform the error checking, but LOG_ERR with value report seems overkill.

etienne-lms committed:
drivers: eeprom: stm32: test HAL return value

Add missing test of HAL_FLASHEx_DATAEEPROM_Unlock() return value.

By the way, add a error trace message when failing to relock the
EEPROM after we failed to program it.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>
